test/wallet-test.js

```diff
@@ -1912,6 +1913,54 @@ describe('Wallet', function() {
     }
   });
 
+  it('should not exceed MEMPOOL_MAX_ANCESTORS policy', async () => {
+    // Create wallet and get one address
+    const wallet = await wdb.create();
+    const addr1 = await wallet.receiveAddress();
+
+    // Dummy address for outputs
+    const recAddr = Address.fromHash(Buffer.alloc(20, 1));
+
+    // Add one single, unconfirmed coin to wallet
+    const mtx1 = new MTX();
+    mtx1.addInput(dummyInput());
+    mtx1.addOutput(addr1, 10 * 1e8);
+    const tx1 = mtx1.toTX();
+    await wallet.txdb.add(tx1, null);
+
+    // Spend unconfirmed change outputs up to the limit
+    const limit = policy.MEMPOOL_MAX_ANCESTORS;
+    for (let i = 0; i < limit; i++) {
+      const tx = await wallet.send({
+        outputs: [{
+          address: recAddr,
+          value: 10000}]
+      });
+      assert(await wallet.txdb.hasPending(tx.hash()));
+    }
+
+    // At the limit
+    const pending = await wallet.getPending();
+    assert(pending.length = policy.MEMPOOL_MAX_ANCESTORS);
+
+    // One more unconfirmed change spend would exceed the limit
+    let err;
+    try {
+      await wallet.send({
+        outputs: [{
+          address: recAddr,
+          value: 10000}]
+      });
+    } catch (e) {
+      err = e;
+    }
+
+    assert(err);
+    assert.strictEqual(
+      err.message,
+      'TX exceeds maximum unconfirmed ancestors.');
+  });
+
   it('should cleanup', async () => {
     consensus.COINBASE_MATURITY = 100;
     // await wdb.close();
```

Comment:
Maybe you could delete this dead code while you are here?

Reply:
Sure... https://github.com/bcoin-org/bcoin/pull/873/commits/1b6a0b62f646c17edfd81a86c31da8f2de9bf15d